presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -378,7 +378,7 @@ public static Boolean stIsClosed(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(BOOLEAN)
     public static Boolean stIsEmpty(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
-        return deserialize(input).isEmpty();
+        return deserializeEnvelope(input) == null;
```

Comment:
Also I'm wondering if this is ok that for the incorrect data (empty input) the method will be returning `false`. Shouldn't it fail instead?  The previous version would fail with VerifyException.

Reply:
@arhimondr I think you are suggesting that `GeometrySerde#deserializeEnvelope` is missing `verify(input.available() > 0);` which is present in `deserialize` and `deserializeType`. Makes sense. I think it should go into the future PR that addresses the design issues you pointed out.

Reply:
> I think it should go into the future PR that addresses the design issues you pointed out.

Yes. But for this very implementation we need to check it in `stIsEmpty` to preserve semantics.